service/src/main/java/com/fecklessweasel/service/objectmodel/FileMetadata.java

```diff
@@ -54,14 +60,19 @@ private FileMetadata(int fid, User user, int course, Date creationDate) {
     public static FileMetadata create(Connection sql,
                                       User user,
                                       int course,
-                                      Date creationDate) throws ServiceException {
+                                      Date creationDate,
+                                      String title,
```

Comment:
You don't check minimum or maximum length anywhere for title or description. This means that the user can just put in an empty title and description for a file. This is very unhelpful. Pick some numbers, define constants, and make sure that they put *something*.

Also, you define title as MAX 64 and description as MAX 255. If you don't check for it *here* and throw proper ServiceException (create new ServiceStatuses for the errors) then when the user makes title too long they will get a generic 500 server encountered an error page.

Finally, most operating systems allow a path of *at least* 255. Some allow up to several thousand. I think file name should maybe be longer than 64, but I leave this up to you.